### PR TITLE
feat(sync): sync can include self url in registry.URLs

### LINKED ISSUE #1335: [Feat]: sync "registry URLs" can include self
### Is your feature request related to a problem? Please describe.

Mainly to achieve a simple (not necessarily consistent) clustering solution so that the configuration becomes simpler.
Let the registry URLs for sync config contain the self address, which should be ignored. So when we have a cluster with many replicas, the same configuration can be applied to all replicas and yet if we push an image to one replica, the other replicas can "sync" from that one.

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by peusebiu**
sync now ignores self referencing urls, this will help in clustering mode where we can have the same config for multiple zots

closes #1335

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/extensions/extension_sync.go [hunk 1]
```diff
@@ -19,6 +23,13 @@ func EnableSyncExtension(config *config.Config, repoDB repodb.RepoDB,
 		onDemand := sync.NewOnDemand(log)
 
 		for _, registryConfig := range config.Extensions.Sync.Registries {
+			// in order to help configure sync in cluster mode, ignore URLs pointing to this server
+			selfAddress := net.JoinHostPort(config.HTTP.Address, config.HTTP.Port)
```
Comment (andaaron) [inline pkg/extensions/extension_sync.go:27]: What if zot is configured to listen in 0.0.0.0? Or in one place of the config it is listed with the hostname and in the other with the IP?

Comment (peusebiu) [reply pkg/extensions/extension_sync.go:27]: yeah, I'm not sure how to solve this...

**Review by andaaron — CHANGES_REQUESTED** [review]
See my other comment.

File: pkg/extensions/extension_sync.go [hunk 2]
```diff
@@ -19,6 +25,19 @@ func EnableSyncExtension(config *config.Config, repoDB repodb.RepoDB,
...
 		for _, registryConfig := range config.Extensions.Sync.Registries {
+			registryConfig := registryConfig
+			if len(registryConfig.URLs) > 1 {
+				if err := removeSelfURLs(config, &registryConfig, log); err != nil {
+					return nil, err
+				}
+			}
+
+			if len(registryConfig.URLs) == 0 {
+				log.Error().Err(zerr.ErrSyncNoURLsLeft).Msg("unable to start sync extensions")
```
Comment (andaaron) [inline pkg/extensions/extension_sync.go:36]:
```suggestion
				log.Error().Err(zerr.ErrSyncNoURLsLeft).Msg("unable to start sync extension")
```

**Review by andaaron — DISMISSED** [review]

**Comment by rchincha** [comment]
Can we add a bats test for this pls? This is user-facing, so we need to be able to point to working examples.

**Comment by rchincha** [comment]
Also, what happens in the k8s case. We can use kind for this.

**Comment by peusebiu** [comment]
> Also, what happens in the k8s case. We can use kind for this.

added bats test for clustering, but for this one can you give more details please?

**Comment by andaaron** [comment]
Let's merge after rebase

**Pull request merged**